spyder/api/plugins/new_api.py

```diff
@@ -1,228 +1,398 @@
+    """
+    The monospace (code) font size delta relative to Spyder default, in points.
+    """
+
+    INTERFACE_FONT_SIZE_DELTA: int = 0
+    """
+    The interface (UI) font size delta relative to Spyder default, in points.
+    """
+
+    CONTEXT_NAME: str | None = None
+    """
+    The name under which to store actions, toolbars, toolbuttons and menus.
+    """
+
+    CAN_BE_DISABLED: bool = True
+    """
+    Define if a plugin can be disabled in the Spyder :guilabel:`Preferences`.
+
+    If ``False``, the plugin is considered "core" and cannot be disabled.
+    ``True`` by default, meaning the plugin can be disabled.
+    """
+
+    REQUIRE_WEB_WIDGETS: bool = False
+    """
+    Declare whether the plugin needs to use Qt web widgets (QtWebEngine).
+
+    Qt Web Widgets is a heavy dependency for many packagers, e.g. Conda-Forge.
+    We thus ask plugins to declare whether or not they need
+    web widgets to make it easier to distribute Spyder without them.
+
+    `More information <https://github.com/spyder-ide/spyder/pull/22196#issuecomment-2189377043>`__
+    """
 
     # --- API: Signals -------------------------------------------------------
     # ------------------------------------------------------------------------
     # Signals here are automatically connected by the Spyder main window and
     # connected to the the respective global actions defined on it.
     sig_free_memory_requested = Signal()
     """
-    This signal can be emitted to request the main application to garbage
-    collect deleted objects.
+    Request the main application garbage collect deleted objects.
     """
 
     sig_plugin_ready = Signal()
     """
-    This signal can be emitted to reflect that the plugin was initialized.
+    Emitted once the plugin is initialized.
     """
 
     sig_quit_requested = Signal()
     """
-    This signal can be emitted to request the main application to quit.
+    Request that the main Spyder application quit.
     """
 
     sig_restart_requested = Signal()
     """
-    This signal can be emitted to request the main application to restart.
+    Request that the main Spyder application perform a restart.
     """
 
     sig_status_message_requested = Signal(str, int)
     """
-    This signal can be emitted to request the main application to display a
-    message in the status bar.
+    Request that the main application display a status bar message.
 
     Parameters
     ----------
     message: str
-        The actual message to display.
+        The message to display.
     timeout: int
-        The timeout before the message disappears.
+        The timeout before the message disappears, in milliseconds.
     """
 
     sig_redirect_stdio_requested = Signal(bool)
     """
-    This signal can be emitted to request the main application to redirect
-    standard output/error when using Open/Save/Browse dialogs within widgets.
+    Request redirection of ``stdout``/``stderr`` when using Open/Save/Browse.
```

Comment:
```suggestion
    Request redirection of ``stdout``/``stderr`` when using Open/Save/Browse
    dialogs in widgets.
```

The current docstring makes no sense (what `Open/Save/Browse` means?)

Reply:
All common docstring standards and conventions (e.g. PEP 257, Numpydoc, Googledoc) require the summary line to fit in one line (and this is relied upon by Sphinx-Autosummary to populate the summary tables, etc).

However, I can do a much better job of shortening this and then adding another line after for more context.

```suggestion
    Request the main app redirect standard out/error within file pickers.
    
    This will redirect :data:`~sys.stdout` and :data:`~sys.stderr` when using
    :guilabel:`Open`, :guilabel:`Save`, and :guilabel:`Browse` dialogs
    within a plugin's widgets.
```